### PR TITLE
supergfxctl-plasmoid: init at 2.0.0

### PR DESCRIPTION
**Pull request by Johny-LPM**
## Description of changes
Adds the [supergfxctl-plasmoid](https://gitlab.com/Jhyub/supergfxctl-plasmoid) to Plasma 6 in order to easily use [supergfxctl](https://gitlab.com/asus-linux/supergfxctl) from the system tray (or anywhere you put the widget)

NOTE: There's no binary to test with this, but I have rebuilt my NixOS system and confirmed it works as intended, like it would on any other system (personally observed the same behavior as on Arch and on Fedora) 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by clebs** [comment]
Currently, the package is not exposed. Please move it to `pkgs/by-name/su/supergfxctl-plasmoid/package.nix`.

(The old way would have been to add the package to `pkgs/top-level/all-packages.nix`, but this pattern is being phased out.)

**Comment by Johny-LPM** [comment]
Oh, got it! Sorry!

**Comment by Johny-LPM** [comment]
Like so?

File: pkgs/by-name/su/supergfxctl-plasmoid/package.nix [hunk 1]
```diff
@@ -0,0 +1,55 @@
+{ lib
+, stdenv
+, fetchFromGitLab
+, cmake
```
Comment (clebs) [inline pkgs/by-name/su/supergfxctl-plasmoid/package.nix:4]: This one might also be already included (can't try it out right now)

File: pkgs/by-name/su/supergfxctl-plasmoid/package.nix [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    kdePackages.full
+    kdePackages.frameworkintegration
+  ];
+
+  buildPhase = ''
+    cmake -DBUILD_WITH_QT6=ON ..
+    make
+  '';
+
+  meta = with lib; {
```
Comment (eclairevoyant) [inline pkgs/by-name/su/supergfxctl-plasmoid/package.nix:45]:
```suggestion
  meta = {
```

tracking issue: #292468
discouraged as it leads to ambiguous `with`s

File: pkgs/by-name/su/supergfxctl-plasmoid/package.nix [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    description = "KDE Plasma plasmoid for supergfxctl";
+    longDescription = ''
+      KDE Plasma plasmoid for supergfxctl
+      Built as a C++/QML Plasmoid
+    '';
+    license = licenses.mpl20;
```
Comment (eclairevoyant) [inline pkgs/by-name/su/supergfxctl-plasmoid/package.nix:51]:
```suggestion
    license = lib.licenses.mpl20;
```

File: pkgs/by-name/su/supergfxctl-plasmoid/package.nix [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    homepage = "https://gitlab.com/Jhyub/supergfxctl-plasmoid";
+    maintainers = with maintainers; [ johnylpm ];
```
Comment (eclairevoyant) [inline pkgs/by-name/su/supergfxctl-plasmoid/package.nix:53]:
```suggestion
    maintainers = with lib.maintainers; [ johnylpm ];
```

File: pkgs/by-name/su/supergfxctl-plasmoid/package.nix [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    kdePackages.kconfig
+    kdePackages.ki18n
+    kdePackages.kirigami
+    kdePackages.ksvg
+    kdePackages.libplasma
+    kdePackages.full
+    kdePackages.frameworkintegration
+  ];
+
+  buildPhase = ''
```
Comment (eclairevoyant) [inline pkgs/by-name/su/supergfxctl-plasmoid/package.nix:40]: `buildPhase` seems superfluous here

File: pkgs/by-name/su/supergfxctl-plasmoid/package.nix [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+}:
+
+stdenv.mkDerivation rec {
+  pname = "supergfxctl-plasmoid";
+  version = "2.0.0";
+
+  src = fetchFromGitLab {
+    owner = "jhyub";
+    repo = pname;
+    rev = "v2.0.0";
```
Comment (eclairevoyant) [inline pkgs/by-name/su/supergfxctl-plasmoid/package.nix:17]:
```suggestion
    rev = "refs/tags/v${version}";
```

File: pkgs/by-name/su/supergfxctl-plasmoid/package.nix [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+    sha256 = "1c9s3mab6p8bbanc1b5ypb5rjl5n9ma32814ra1amdpxa1n6cwwv";
```
Comment (eclairevoyant) [inline pkgs/by-name/su/supergfxctl-plasmoid/package.nix:18]:
```suggestion
    hash = "sha256-m3NmbFD9tqqCyiQgMVRNtlCZy7q+rMCsWgtds1QdOrE=";
```

SRI hash preferred

**Review by eclairevoyant — COMMENTED** [review]
Thanks for the contribution! Some feedback below.
Also you will eventually want to rebase + squash into two commits, the `maintainers: ...` and the package init

File: pkgs/by-name/su/supergfxctl-plasmoid/package.nix (hunk 5 above)
Comment (Gerg-L) [reply pkgs/by-name/su/supergfxctl-plasmoid/package.nix:40]: I agree setting `makeFlags` should be more than enough

File: pkgs/by-name/su/supergfxctl-plasmoid/package.nix [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+  };
+
+  nativeBuildInputs = [
+    cmake
+    kdePackages.extra-cmake-modules
+    gcc
```
Comment (Gerg-L) [inline pkgs/by-name/su/supergfxctl-plasmoid/package.nix:24]:
```suggestion
```
gcc is in stdenv by default

**Comment by Johny-LPM** [comment]
Thank you so much for all your comments! As you can see, it's my first time contributing, so I'll be sure to keep all your suggestions in mind for the future. 
That being said, I wasn't aware gcc was part of stdenv, so that's off. Didn't know about the "pname" issue, many of my sources were packages I looked up that are already present in the repos that also used the same structure, but understood! I'll also be sure to keep in mind the better practice for hashes and revisions in mind. 
Furthermore, indeed the buildPhase wasn't necessary, but makeFlags (which I actually wasn't even aware of, as an option) didn't appear to be needed either! I've now tested without either and it seems that the defaults are fine. Cmake is still needed though.
Moreover, after other packages I've since been experimenting with had different deps needed, some more necessary and others unnecessary from upstream projects, I've further tested with removing packages from the buildInputs until I arrived at what seems to be the minimal amount, shaving off the rest that seemingly wasn't actually required.
In any case, thank you so much for everything! If there are any further suggestions for improving, be it the package or just the practices for packaging software, I'll be sure to absorb as much as possible!

File: pkgs/by-name/su/supergfxctl-plasmoid/package.nix [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+    rev = "refs/tags/v${version}";
+    hash = "sha256-m3NmbFD9tqqCyiQgMVRNtlCZy7q+rMCsWgtds1QdOrE=";
+  };
+
+  nativeBuildInputs = [
+    cmake
+  ];
+
+  buildInputs = [
+    kdePackages.wrapQtAppsHook
```
Comment (eclairevoyant) [inline pkgs/by-name/su/supergfxctl-plasmoid/package.nix:24]: Just noticed this, but this should go in `nativeBuildInputs`

**Review by eclairevoyant — COMMENTED** [review]
Also, please rebase + squash into the two commits mentioned earlier

File: pkgs/by-name/su/supergfxctl-plasmoid/package.nix (hunk 9 above)
Comment (Johny-LPM) [reply pkgs/by-name/su/supergfxctl-plasmoid/package.nix:24]: gotcha, I'll correct that as well
also, sorry, I'm trying to look into the rebasing and squashing to try and get it right, my bad

Comment (Johny-LPM) [reply pkgs/by-name/su/supergfxctl-plasmoid/package.nix:59]: I'm having some trouble with the whole commit squashing
if it's not too much trouble, if and when at all possible, could you help me out?

**Comment by eclairevoyant** [comment]
`git rebase -i` should give open an editor window:

* delete the lines you don't want to include (merge commits should be dropped)
* keep the lines you do want to include (the two commits you intend to keep)
* reorder the lines to reorder the commits. the first commit is at the top.
* replace "pick" with "s" or "squash" to squash the commit into the previous line (this is useful for if you have changes across multiple commits that you want to combine into one big commit)
* then save and exit the editor

A more detailed guide is available at https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History

**Comment by Johny-LPM** [comment]
Does that do it? I'm not really sure, the fact that I made the mistake of merging during the PR and having changed the file location threw me off during rebasing

**Comment by SuperSandro2000** [comment]
just going to squash merge this

**Pull request merged**